code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -77,14 +77,16 @@
 	id = "vapor"
 
 /datum/gas_reaction/water_vapor/init_reqs()
-	requirements = list(/datum/gas/water_vapor = MOLES_GAS_VISIBLE)
+	requirements = list(
+		/datum/gas/water_vapor = MOLES_GAS_VISIBLE,
+		"MAX_TEMP" = WATER_VAPOR_CONDENSATION_POINT,
+	)
 
-/datum/gas_reaction/water_vapor/react(datum/gas_mixture/air, datum/holder)
+/datum/gas_reaction/water_vapor/react(datum/gas_mixture/air, turf/open/location)
```

Comment:
keep the default typing, it's called holder at base, shouldn't rename it just here